src/qudi/logic/pulsed/pulse_objects.py

```diff
@@ -1641,5 +1642,34 @@ def is_generator_class(obj):
         @return bool: True if obj is a valid generator class, False otherwise
         """
         if inspect.isclass(obj):
-            return PredefinedGeneratorBase in obj.__bases__ and len(obj.__bases__) == 1
+            return PredefinedGeneratorBase in obj.__bases__# and len(obj.__bases__) == 1
         return False
+
+    def activate_plugins(self):
+        [gen.activate_plugin() for gen in self._generator_instances if hasattr(gen, 'activate_plugin')]
+
+
+class PredefinedGeneratorPlugin():
+    """
+    PredefinedGeneratorPlugin is a PredefinedGenerator with addtional powers.
+    - It may manipulse the sequence generation parameters from it's code
+    - It can run code after the PulseObjectGenerator in order to manipulate all loaded predefined methods.
+    """
+    def activate_plugin(self):
+        # allow plugins to invoke code after the PulseObjectGenerator is fully initialized
+        pass
+
+    @property
+    def generation_parameters(self):
```

Comment:
Maybe one could add some kind of interface method to `PredefinedGeneratorBase` to get/insert parameters from/into the `SequenceGeneratorLogic`.